### PR TITLE
Make amqp less fragile

### PR DESCRIPTION
**Pull request by coolo**
Instead of keeping an AMQP connection alive all the time (and risk losing it
on subprocess forks), just create a Publisher on events and fire it

See  https://progress.opensuse.org/issues/40652

### CODE REVIEW

**Comment by coolo** [comment]
I'm already fixing the dev container

**Comment by asdil12** [comment]
Yes - this works, because `pubsub` is actually declared in a permanent way (_durable_) on the server side (see https://github.com/openSUSE/suse_msg/blob/master/amqp_infra.md#the-amqp-server).

This is why I just did a passive declare which is essentially optional but just checks if the exchange already exists on the server (and has the _durable_ flag).

**Comment by Martchus** [comment]
```
Found Mojo::RabbitMQ::Client v0.2.1 which doesn't satisfy 0.2.
```

**Comment by coolo** [comment]
Now seriously - who puts a v into the version?

File: cpanfile [hunk 1]
```diff
@@ -53,7 +53,7 @@ requires 'Mojo::IOLoop';
 requires 'Mojo::IOLoop::ReadWriteProcess', '0.20';
 requires 'Mojo::JSON';
 requires 'Mojo::Pg';
-requires 'Mojo::RabbitMQ::Client';
+requires 'Mojo::RabbitMQ::Client', '>= v0.2';
```
Comment (SPodjasek) [inline cpanfile]: I'm curius too when did that 'v' prefix come to life on cpanmetadb

**Pull request merged**